training/security.md

```diff
@@ -0,0 +1,161 @@
+---
+title: Secure Your Cluster
+summary: A secure CockroachDB cluster uses TLS for encrypted inter-node and client-node communication and requires CA, node, and client certificates and keys.
+toc: false
+sidebar_data: sidebar-data-training.json
+---
+
+<div id="toc"></div>
+
+## Presentation
+
+<iframe src="https://docs.google.com/presentation/d/e/2PACX-1vSozBu9RIF-1V73oPtK-HTJpdtG3euC4oS8aR2Ze0AmIkBcTr1pKjblJ9Q6TJBLbREEX8wmcRkzrLeq/embed?start=false&loop=false" frameborder="0" width="756" height="454" allowfullscreen="true" mozallowfullscreen="true" webkitallowfullscreen="true"></iframe>
+
+#### URL for comments
+
+[Secure Your Cluster](https://docs.google.com/presentation/d/1RqiRikOPMHdtynASNdl20Vfxxd764vuNFRaxCVeXLB4/)
+
+## Lab
+
+Up to this point, we've been using an insecure local cluster. However, when using CockroachDB on remote servers, it's necessary to secure your deployment––when CockroachDB runs in insecure mode, it's _really_ insecure.
+
+In this lab, we'll show you how to secure your cluster.
+
+### Before You Begin
+
+To complete this lab, you need to have the [`cockroach` binary installed](/stable/install-cockroachdb.html).
+
+We also assume you have a local, insecure 3-node cluster running. If you don't, you can skip the **Tear down existing cluster** section of this lab.
+
+### Overview
+
+1. Tear down existing cluster.
+2. Generate certificate files.
+3. Restart cluster with SSL encryption.
+4. Connect to the cluster using encryption.
+
+### Step 1. Tear Down Cluster
+
+You can do this elegantly by using `cockroach quit` on each node, but we're going to close down everything in one fell swoop:
+
+~~~ shell
+$ pkill -9 cockroach
+~~~
+
+### Step 2. Generate Certificate Files
+
+We're going to generate:
+
+- A certificate authority (CA) to sign all certificates
+- One node certificate per CockroachDB node
+- One client cert per client
+
+#### Certificate Authority
+
+~~~ shell
+$ mkdir certs
+
+$ mkdir my-safe-directory
+~~~
+
+We want to ensure `my-safe-directory`, which will ultimately contain your CA key, is secure, so we should also modify its permissions:
+
+~~~
+chown my-safe-directory
```

Comment:
nit: add `$` prompt

Reply:
Also, as-is this command doesn't do anything for me:

```
~/crdb-training$ chown my-safe-directory
usage: chown [-fhv] [-R [-H | -L | -P]] owner[:group] file ...
       chown [-fhv] [-R [-H | -L | -P]] :group file ...
```

Reply:
Fixed and fixed.